civismlext/hyperband.py

```diff
@@ -481,16 +482,19 @@ def _process_outputs(self, out, n_splits):
         # Use one MaskedArray and mask all the places where the param is not
         # applicable for that candidate. Use defaultdict as each candidate may
         # not contain all the params
-        param_results = defaultdict(partial(MaskedArray,
-                                            np.empty(n_candidates,),
-                                            mask=True,
-                                            dtype=object))
-        for cand_i, params in enumerate(candidate_params):
+        param_results = {}
+        for cand_idx, params in enumerate(candidate_params):
             for name, value in params.items():
                 # An all masked empty array gets created for the key
                 # `"param_%s" % name` at the first occurence of `name`.
                 # Setting the value at an index also unmasks that index
-                param_results["param_%s" % name][cand_i] = value
+
+                param_name = "param_" + name
+                if param_name not in param_results:
+                    # The array of candidates-to-values not yet definied for this parameter
+                    param_results[param_name] = np.ma.MaskedArray(np.empty(n_candidates,), mask=True, dtype=object, fill_value=-1)
```

Comment:
Why did you put `fill_value=-1` here?  Did leaving it as the default not work anymore?

Reply:
Also, to your question in #53, yeah, I think it'd be better to use `np.zeros` instead of `np.empty`, though I don't think it matters in practice here because the values will either be masked or replaced in this loop.

Reply:
Oh, I wasn't sure if the fill_value mattered, and thought I'd make it clear that they were missing to the user (in case of in-depth analysis or debugging), but I can just set it to default if you feel that's more fitting.

Reply:
I don't have a strong feeling, but my inclination would be to leave it as is.  It looks like the masked array gets returned as a masked array rather than filled (and then eventually attached to the object as `cv_results_`), so the fill value doesn't really matter. Also, the default fill value for `dtype=object` is `'?'` as far as I can tell, which seems like a reasonable indicator of missingness. 🤷‍♂️

Reply:
Ok, I'll just let it set a default value then, no worries!